### PR TITLE
Metrics-Webapp: Implement React components, Redux store, routing, styling and testing.

### PR DESCRIPTION
**Pull request by kessie2862**
## Description
This PR adds several key components and configurations to the project. The changes include

- [ ] Added React components: `DetailsPage`, `Filter`, and `HomePage`.
- [ ] Implemented `Redux store` with categories slice for managing category data.
- [ ] Created routing using `react-router-dom` for navigating between different pages.
- [ ] Configured the `Redux store` and combined reducers using `@reduxjs/toolkit.`
- [ ] Rendered the React application to the root element in the HTML document.
- [ ] Added styles to improve the `UI`
- [ ] Add icon to the `details` page to navigate back to `homepage`
- [ ] Installed icons packages
- [ ] Added `footer` to the application
- [ ] Implemented unit tests for pure functions within the Redux codebase.
- [ ] Developed integration tests utilizing the React Testing Library.
- [ ] Implemented `API mocking` to isolate the tests from actual network requests.
- [ ] Mocked the connection to the `Redux` Store in the tests.
- [ ] Addressed and resolved all `ESLint` linter errors to adhere to the defined code style guidelines.

### CODE REVIEW

File: src/components/styles/DetailsPage.css [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+  left: 2rem;
+}
+
+.left-arrow-icon {
+  font-size: 2rem;
+}
+
+.card {
+  background-color: #fff;
+  border-radius: 0.5rem;
```
Comment (joskalenda) [inline src/components/styles/DetailsPage.css:34]: - [ ] As per the design layout, please make sure you adjust the background color for your n + 1 items, I mean in the original design when the 1st card is black, the following is gray, and so on, I think the following image explains what I mean by this



Hint: `card:nth-child( odd ) { backgound-color: gray;}`

File: src/components/styles/DetailsPage.css [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+  text-align: center;
+}
+
+.card img {
+  max-width: 100px;
+  margin-bottom: 1rem;
+}
+
+.card p {
+  margin-bottom: 1rem;
```
Comment (joskalenda) [inline src/components/styles/DetailsPage.css:48]: - [ ] Same as here, please make sure you follow the layout, either you can change the color to your preferred one but the page wireframe must be identic to the design

**Review by joskalenda — CHANGES_REQUESTED** [review]
| STATUS: REQUIRED CHANGES ♻️  |
|--------------------------------|
| Hi @kessie2862 , |
|Good job so far! There are some issues that you still need to work on to go to the next project but you are almost there! |
|  |

## Highlights

- [ ]... ✅ Great design
- [ ].. ✅ Green linters
- [ ].... ✅ Great readme file

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

-------
> Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me(@joskalenda) in your question so I can receive the notification.

> **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/styles/DetailsPage.css (hunk 1 above)
Comment (kessie2862) [reply src/components/styles/DetailsPage.css:34]: Thanks for the feedback and hint. I have implemented the changes 👍

File: src/components/styles/DetailsPage.css (hunk 2 above)
Comment (kessie2862) [reply src/components/styles/DetailsPage.css:48]: I have made the necessary changes. Thanks 😊

**Review by DuaneDave — APPROVED** [review]
Hi @kessie2862 👋, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
✔️ Your web app is accessible due to the search bar
✔️ The styling looks good

## Optional suggestions
N/A

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code. Some of them were missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**